tracing/trace_state.hh

```diff
@@ -286,11 +286,6 @@ private:
         _records->session_rec.started_at = std::chrono::system_clock::now();
     }
 
-    template <typename Func>
-    void begin(const seastar::lazy_eval<Func>& lf, gms::inet_address client) {
-        begin(lf(), client);
-    }
-
```

Comment:
I don't see that anything is sacrificed. The lazy_eval thing was unlazied immediately.

Reply:
Yes. This version of a call used to delay string formatting to the point when the `tracing::begin()` global static wrapper may detect if `trace_state_ptr` is initialized and only then do the evaluation. As far as we no more support strings in `tracing::begin` we are deleting this.